lib/contentful/resource/custom_resource.rb

```diff
@@ -0,0 +1,19 @@
+module Contentful
+  module Resource
+    module CustomResource
+      def initialize(*)
+        super
+
+        unless client.nil?
+          if client.dynamic_entry_cache.key?(content_type['sys']['id'].to_sym)
```

Comment:
@neonichu The entry will still have fields, just not dynamically assigned.

And can be read via `entry.fields[:field_name]`

Also, the only case when `client == nil` is when you create the entry manually like `MyCustomEntry.new`, in which case you can provide a hash as a parameter with the values for the fields, which will be available even if the CT is not cached.

Only possible exception is: Client exists, CT not cached, that case can happen under 2 conditions:
1) Client created before CT was and no client reload happened
2) More than 1000 CTs in the Space and entry was requested by ID and is of one of the CTs left out